pages/jobs/[slug].tsx

```diff
@@ -0,0 +1,35 @@
+import type { NextPage } from 'next';
+import { getJobPost, getAllJobSlugs } from 'lib/peopleHR';
+import type { JobPost } from 'lib/peopleHR';
+import styles from 'styles/Jobs.module.scss';
+
+const JobPosting: NextPage<{ job: JobPost }> = ({ job }) => {
+    return (
+        <div className={styles.container}>
+            <h1>{job.title}</h1>
+            <div
+                dangerouslySetInnerHTML={{
+                    __html: job.description,
+                }}
+            ></div>
+        </div>
+    );
+};
+
+export default JobPosting;
+
+export async function getStaticProps({ params }: { params: { slug: string } }) {
+    const job = (await getJobPost(params.slug)) ?? [];
```

Comment:
**Chore:** We could throw a 404 here if there are no matches .
```
    if (!job) {
        return {
        notFound: true,
        }
    }
```

Reply:
We would probably want to follow Thibauds example [here](https://git.torchbox.com/thibaudcolas/llamanext/-/blob/main/pages/blog/%5Bslug%5D.tsx#L30) with `fallback: false` in getStaticPaths

Reply:
To prevent a query being sent to peopleHR for every URL a user enters, we should use `fallback: false` and trigger a fresh build of the website via a cron job instead. This could run every hour (744 builds a month) and we'd have spare builds left over for development (1,300 to 2,300, depending on how we trigger the build)

Reply:
If ISR already handles generating new job postings can't we just rely on `revalidate` instead? Why the need of a cron job or was this comment before we tested that ISR could handle that.

Reply:
Instead of sending a query to get the PeopleHR feed for every incorrect URL entered (potentially DDOS'ing the feed), I've implemented a simple caching system so we only poll PeopleHR once and refer to the cached data from then on. I've currently set the cache TTL to ten seconds for testing, for production I'd suggest setting the cache to 60 minutes (the same as our ISR revalidation). Adding caching gives the added bonus of Next.js not querying the feed 20+ times during build